yt_dlp/extractor/facebook.py

```diff
@@ -1017,19 +1034,31 @@ def _real_extract(self, url):
         video_id = self._match_id(url)
         webpage = self._download_webpage(url, video_id)
 
-        post_data = [self._parse_json(j, video_id, fatal=False)
-                     for j in re.findall(r's\.handle\(({.*})\);requireLazy\(', webpage)]
-        data = traverse_obj(post_data, (
-            ..., 'require', ..., ..., ..., 'props', 'deeplinkAdCard', 'snapshot', {dict}), get_all=False)
+        if post_data := [self._parse_json(j, video_id, fatal=False)
+                         for j in re.findall(r'data-sjs>({.*?ScheduledServerJS.*?})</script>', webpage)]:
+            data = traverse_obj(post_data, (
+                ..., 'require', ..., ..., ..., '__bbox', 'require', ..., ..., ...,
+                'entryPointRoot', 'otherProps', 'deeplinkAdCard', 'snapshot', {dict}), get_all=False)
+        elif post_data := [self._parse_json(j, video_id, fatal=False)
+                           for j in re.findall(r's\.handle\(({.*})\);requireLazy\(', webpage)]:
+            data = traverse_obj(post_data, (
+                ..., 'require', ..., ..., ..., 'props', 'deeplinkAdCard', 'snapshot', {dict}), get_all=False)
         if not data:
             raise ExtractorError('Unable to extract ad data')
 
         title = data.get('title')
         if not title or title == '{{product.name}}':
             title = join_nonempty('display_format', 'page_name', delim=' by ', from_dict=data)
+        markup = None
+        if markup_id := traverse_obj(data, ('body', '__m', {str_or_none}), get_all=False):
+            markup = clean_html(traverse_obj(post_data, (
+                ..., 'require', ..., ..., ..., '__bbox', 'markup', lambda _, v: v[0].startswith(markup_id),
+                ..., '__html', {lambda x: x if x and not x.startswith('&#123;&#123;product.') else None}),
+                get_all=False))
 
         info_dict = traverse_obj(data, {
-            'description': ('link_description', {str}, {lambda x: x if x != '{{product.description}}' else None}),
+            'title': ({lambda x: title}),
+            'description': ('link_description', {lambda x: markup or (x if x and not x.startswith('{{product.') else None)}),
             'uploader': ('page_name', {str}),
             'uploader_id': ('page_id', {str_or_none}),
             'uploader_url': ('page_profile_uri', {url_or_none}),
```

Comment:
I'm not a big fan of overusing transformations (e.g. `lambda x: title`). IMO it would cleaner to leverage `merge_dicts` here:
```py
        info_dict = merge_dicts({
            'title': title,
            'description': markup or None,
        }, traverse_obj(data, {
            'description': ('link_description', {lambda x: x if not x.startswith('{{product.') else None}),
            'uploader': ('page_name', {str}),
            'uploader_id': ('page_id', {str_or_none}),
            'uploader_url': ('page_profile_uri', {url_or_none}),
            'timestamp': ('creation_time', {int_or_none}),
            'like_count': ('page_like_count', {int_or_none}),
        }))
```